bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java

```diff
@@ -825,20 +825,20 @@ private void resetSize(int index) {
         }
     }
 
-    private void validateTupleSizeAndInherentType() {
+    private void validateTupleSizeAndInherentType(long index) {
```

Comment:
Since we cast the index to `int` in multiple places, we can pass the index as `int` and cast it when we call the `validateTupleSizeAndInherentType`.

Reply:
I thought we supported large values for indexing in tuples and the casting would make it erroneous. I was wrong. I'll change that. Thank you for pointing that out.